search/task_helpers.py

```diff
@@ -355,6 +358,37 @@ def delete_course(course_obj):
     delete_document.delay(
         gen_course_id(course_obj.platform, course_obj.course_id), COURSE_TYPE
     )
+    for content_file in ContentFile.objects.filter(run__object_id=course_obj.id).filter(
+        run__content_type=ContentType.objects.get(model=COURSE_TYPE)
+    ):
+        delete_content_file(content_file)
+
+
+@if_feature_enabled(INDEX_UPDATES)
```

Comment:
We have #2521 open for removing this flag, so I think you can just drop it here and on the one below